src/constants/sharepoint.constant.ts

```diff
@@ -1,5 +1,10 @@
+import { DOCUMENT_FILE_TYPE } from './document-file-type.constant';
+
 export const SHAREPOINT = {
   RESOURCE_NAME: {
     REGEX: /^[\w\-.()\s]+$/,
   },
+  DOCUMENT_FILE_NAME: {
+    REGEX: new RegExp(`^(?!\\s)[\\w\\-.()\\s]+\.(${Object.values(DOCUMENT_FILE_TYPE).join('|')})(?<![\\s.])$`, 'i'),
```

Comment:
GitHub is commenting `Found non-literal argument to RegExp Constructor` on this line, but we should be fine here because there is no user input used in the RegExp